### PR TITLE
feat(website): Hero community section

### PR DESCRIPTION
**Pull request by vasucp1207**
## Summary

- Related to https://github.com/biomejs/biome/issues/882

- Inspired by the astro docs.

- The community section currently feels a bit empty to me(left side of the section and on small screens) as of now, any suggestions on how we could do this better, or is it okay?

### CODE REVIEW

**Comment by ematipico** [comment]
@vasucp1207 I wish you would start filling out the PR template. Many contributors - even first-time contributors - make the effort to provide some information. Could you please do the same?

**Comment by ematipico** [comment]
Thank you @vasucp1207 ! 

This is an amazing addition to our homepage. 

> The community section currently feels a bit empty to me(left side of the section and on small screens) as of now, any suggestions on how we could do this better, or is it okay?

I think it's OK! However, I think we should show the Github avatars on tablets and mobile phones too, and maybe we can blur them if they happen to go under the buttons. Check how the Astro website does it. This way, it doesn't feel empty anymore.

Here's more feedback: 
- I am not sure what "Powered by Astro" means here. Do you mean that the **website** is powered by Astro? If so, I don't think we should mention it. Maybe just saying "Powered by our open source contributors" should be enough 
- I think we should change how we fetch and show the GitHub avatars. On this page, I think we should sort the contributors by their number of contributors (very much like the order of contributors you see here: https://github.com/biomejs/biome/graphs/contributors), and render them **statically**, so we don't ship any additional JavaScript in the homepage. 
- I noticed that the homepage downloads a component that we don't use in the homepage. Could you check why? Maybe I am wrong and we do use it somewhere
Screenshot 2023-12-22 at 11 17 36

File: website/src/frontend-scripts/contributors.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const applyStyles = (element: any, styles: any) => {
```
Comment (victor-teles) [inline website/src/frontend-scripts/contributors.ts:1]: Seems this code is not properly formatted, you can format it using `pnpm format`

File: website/src/components/Community.astro [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+import Contributors from "@src/components/Contributors.astro";
+import { Icon } from "@astrojs/starlight/components";
+---
+
+<div class="community">
+  <div class="community-grid"></div>
+  <div class="community-head">Community</div>
```
Comment (victor-teles) [inline website/src/components/Community.astro:8]: The contrast ratio for this title and the description seems a little bit lower, you should keep some similarity to "Biome is fast" title

File: website/src/components/Community.astro [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="community-connect">
+    <div class="community-description">Powered by Astro and our open-source contributors</div>
+    <div class="community-links">
+      <a class="discord">
+        <Icon name="discord" /> Discord
```
Comment (victor-teles) [inline website/src/components/Community.astro:13]: This links are missing the `href`

**Review by victor-teles — CHANGES_REQUESTED** [review]

**Comment by victor-teles** [comment]
IMO this should also be available in other languages, even in English.
Any thoughts?

By the way, the way the component was made make translation a bit hard
@vasucp1207 Maybe you need to rethink how to support the translation to this component

**Comment by vasucp1207** [comment]
> * I noticed that the homepage downloads a component that we don't use in the homepage. Could you check why? Maybe I am wrong and we do use it somewhere
>   Screenshot 2023-12-22 at 11 17 36

This also downloads in Starlight and Astro docs.

**Comment by ematipico** [comment]
Good to know, it could be a bug

**Comment by ematipico** [comment]
I found two problems:
- the avatars are incorrectly placed
- the performance banner doesn't have the same padding of the cards

File: website/src/content/docs/index.mdx [hunk 4]
```diff
@@ -20,40 +23,150 @@ hero:
...
+      </a>
+    </div>
+  </div>
+  <div class="connect">
+    <Community />
+  </div>
+</div>
 
+<h2 class="sponsor-head">Sponsors</h2>
 <div class="sponsors">
```
Comment (unvalley) [inline website/src/content/docs/index.mdx:164]: `<h2 class="sponsor-head">Sponsors</h2>` should be  in `<div class="sponsors">`

File: website/src/content/docs/ja/index.mdx [hunk 5]
```diff
@@ -2,15 +2,18 @@
...
-    title: Web開発のためのたった1つのツールチェーン
-    tagline: format、lintなど一瞬で完了させます！
-    image:
-        alt: Biome, ウェブツールチェイン
-        file: ../../../assets/biome-logo-slogan.svg
-    actions:
+  title: Web開発のためのたった1つのツールチェーン
+  tagline: format、lintなど一瞬で完了させます！
+  image:
+    alt: Biome，Web 工具链
```
Comment (unvalley) [inline website/src/content/docs/ja/index.mdx:11]: use Japanese correctly
```suggestion
    alt: Biome, ウェブツールチェイン
```

File: website/public/img/site.webmanifest
Comment (unvalley) [inline website/public/img/site.webmanifest:1]: We don't need this file?

File: website/src/content/docs/index.mdx [hunk 6]
```diff
@@ -20,40 +23,150 @@ hero:
...
-import { Card, CardGrid } from '@astrojs/starlight/components';
+<body>
+  <div class="gradient"></div>
+  <div class="grid"></div>
+</body>
 
-<CardGrid stagger>
+<div class="hero-code">
+  <div class="formatter">
+    <h2 class="head">Biome is fast</h2>
```
Comment (unvalley) [inline website/src/content/docs/index.mdx:47]: Picked this up from Rome's one.
```suggestion
    <h2 class="supercharge-head">Supercharge your workflow</h2>
    <p class="supercharge-description">Experience the seamless integration of a vertical toolchain. Full of useful features like formatting and linting, with even more in the future.</p>
```

**Comment by unvalley** [comment]
Thank you for your contribution. Overall I'd like to take over the Rome's web page. Adding contributors section is good.
Sorry for the poor explanation, but I want to take care of the margins.

File: website/src/playground/components/Progress.tsx
Comment (unvalley) [inline website/src/playground/components/Progress.tsx:1]: The end of the progress should arrive at the right end. With the current progress bar, it is difficult to visually recognize Biome as fast. I attached Rome's one.
Screenshot 2023-12-24 at 17 03 54

**Comment by ematipico** [comment]
@vasucp1207 

I resolved a bunch of conflicts and made a few changes. Regarding the avatars, I removed the style that hides contributors' avatars. I believe it's best to create predefined roles with a predefined number of elements per row, and we should not use CSS to hide the avatars.

**Comment by vasucp1207** [comment]
> Thank you for your contribution. Overall I'd like to take over the [Rome's web page](https://web.archive.org/web/20230203060333/https://rome.tools/). Adding contributors section is good. Sorry for the poor explanation, but I want to take care of the margins.

Feel free to make any changes

**Comment by ematipico** [comment]
@vasucp1207 

That's what I see on iPad
Screen Shot 2023-12-25 at 10 58 35

**Comment by ematipico** [comment]
hey @victor-teles , I am going to merge this PR. It would be best to propose something about the translations.

I checked your comment and it wasn't very actionable. It seems you know what you want, but even I couldn't understand your suggestion.

**Pull request merged**